### DISCUSSION TITLE
Kolom tidak ditemukan ketika menggunakan addMarkers pada leafletProxy

### DISCUSSION THREAD

**Discussion by MF-Faqih** [teamalgoritma/community#684] [Q&A] [answered]
Saya membuat sebuah map interaktif menggunakan leaflet dan leafletproxy di shiny berdasarkan input dari UI, untuk membuat data frame berhasil menerima input saya menggunakan fungsi reacitve sebagaimana kode dibawah

```
 map_df <- reactive({
    apartment_map %>% 
      filter(Price >= input$price[1] & Price <= input$price[2],
             Bedroom == input$bedroom,
             Bathroom == input$bathroom,
             NAME_2 == input$district)
  })
```

map_df merupakan data bertipe sf yang sebelumnya saya terapkan fungsi "sf::st_centroid" untuk mendapatkan centroid dari data (saya tidak bisa menggunakan addMarkers pada tipe data FS dengan kolom geometry), kemudian untuk membuat peta statis saya mengguankan kode:

```
  output$map <- renderLeaflet({
    leaflet() %>% 
      addProviderTiles("CartoDB.Positron") %>% 
      setView(lng = 106.8456, lat = -6.2088,zoom = 11)
      
  })
```

dan untuk membuat map interaktif saya menggunakan kode:

```
  observe({
    leafletProxy("map", data = map_df()) %>% 
      clearMarkers() %>% 
      addMarkers(data = map_df()$centroid,
                 popup = paste0(
                   "<strong>Apartment Found: </strong>",
                   max(Price)
                 ))
  })
```

ketika di run, muncul pesan "Error in paste0: object 'Price' not found", saya tidak mengerti kenapa pesan ini muncul padahal pada data map_df terdapat kolom dengan nama "Price"

**Comment by jafarhf1** [comment]
Dear @MF-Faqih,

Terima kasih atas pertanyaannya, akan kami coba telusuri ya Pak permasalahannya

Terima kasih

**Reply by jafarhf1** [reply, marked as answer]
Dear @MF-Faqih,

Kalau dilihat dari kodenya , karena Bapak memasukkan parameter data pada bagian fungsi `addMarkers(data=map_df()$centroid)` nya maka itu akan mengganti data yang sudah ditentukan pada fungsi `leafletProxy(data=map_df())`. Artinya adalah data yang ada di dalam addMarkersnya sudah bukan `map_df()` melainkan `map_df()$centroid`.  Apakah sudah dicoba dengan `max(map_df()$Price)` ?

Semoga membantu.
